include/filters/median.hpp

```diff
@@ -63,201 +61,190 @@ namespace filters
   Physical description: 366 p.
   Series: Prentice-Hall Series in Automatic Computation
   ---------------------------------------------------------------------------*/
-template <typename elem_type>
+template<typename elem_type>
 elem_type kth_smallest(elem_type a[], int n, int k)
 {
-  register int i,j,l,m ;
-  register elem_type x ;
-  l=0 ; m=n-1 ;
-  while (l<m) {
-    x=a[k] ;
-    i=l ;
-    j=m ;
+  register int i, j, l, m;
+  register elem_type x;
+  l = 0; m = n - 1;
+  while (l < m) {
+    x = a[k];
+    i = l;
+    j = m;
     do {
-      while (a[i]<x) i++ ;
-      while (x<a[j]) j-- ;
-      if (i<=j) {
-        ELEM_SWAP(a[i],a[j]) ;
-        i++ ; j-- ;
+      while (a[i] < x) {i++;}
+      while (x < a[j]) {j--;}
+      if (i <= j) {
+        ELEM_SWAP(a[i], a[j]);
+        i++; j--;
       }
-    } while (i<=j) ;
-    if (j<k) l=i ;
-    if (k<i) m=j ;
+    } while (i <= j);
+    if (j < k) {l = i;}
+    if (k < i) {m = j;}
   }
-  return a[k] ;
+  return a[k];
 }
-#define median(a,n) kth_smallest(a,n,(((n)&1)?((n)/2):(((n)/2)-1)))
+#define median(a, n) kth_smallest(a, n, (((n) & 1) ? ((n) / 2) : (((n) / 2) - 1)))
```

Comment:
Nit: it would be nice for this to be a method, rather than a macro (type-checking and all of that).

Reply:
Done in https://github.com/sloretz/filters/pull/1/commits/143d7f1e60b4fed049485b8ab6c86eb7928c25c5.